--- digikam-4.5.0/CMakeLists.txt~	2014-11-12 11:50:30.000000000 +0100
+++ digikam-4.5.0/CMakeLists.txt	2014-12-11 22:20:42.154363495 +0100
@@ -115,15 +115,6 @@
     ENABLE_TESTING()
 endif()
 
-# Make sure that the FindXXX modules of the local libraries are found
-if(ENABLE_PRIVATE_SHAREDLIBS)
-    # prepend our own FindXXX modules to CMAKE_MODULE_PATH
-    set(CMAKE_MODULE_PATH
-        ${CMAKE_SOURCE_DIR}/extra/libkdcraw/cmake/modules
-        ${CMAKE_MODULE_PATH}
-       )
-endif()
-
 # prepend our own FindXXX modules to CMAKE_MODULE_PATH
 set(CMAKE_MODULE_PATH
     ${CMAKE_SOURCE_DIR}/extra/libkgeomap/cmake/modules
@@ -135,11 +126,7 @@
 
 if(ENABLE_PRIVATE_SHAREDLIBS)
     # from kdegraphics/libs
-    set(KIPI_LOCAL_DIR   extra/libkipi)
-    set(KEXIV2_LOCAL_DIR extra/libkexiv2)
-    set(KDCRAW_LOCAL_DIR extra/libkdcraw)
     set(KFACE_LOCAL_DIR  extra/libkface)
-    set(KSANE_LOCAL_DIR  extra/libksane)
 
     #from extragear/libs
     set(KGEOMAP_LOCAL_DIR       extra/libkgeomap)
--- digikam-4.5.0/extra/CMakeLists.txt~	2014-11-12 11:50:30.000000000 +0100
+++ digikam-4.5.0/extra/CMakeLists.txt	2014-12-11 22:24:58.557737363 +0100
@@ -9,12 +9,7 @@
 
 if(${ENABLE_PRIVATE_SHAREDLIBS})
 
-    # from kdegraphics/libs
-    add_subdirectory(libkipi)
-    add_subdirectory(libkexiv2)
-    add_subdirectory(libkdcraw)
     add_subdirectory(libkface)
-    MACRO_OPTIONAL_ADD_SUBDIRECTORY(libksane)
 
     # from extragear/libs
     add_subdirectory(libkgeomap)
