--- digikam-0.9.4/digikam/libs/dimg/loaders/pngloader.cpp~	2010-03-01 10:26:46.905217555 +0200
+++ digikam-0.9.4/digikam/libs/dimg/loaders/pngloader.cpp	2010-03-01 10:26:49.241660222 +0200
@@ -88,7 +88,7 @@
     unsigned char buf[PNG_BYTES_TO_CHECK];
 
     fread(buf, 1, PNG_BYTES_TO_CHECK, f);
-    if (!png_check_sig(buf, PNG_BYTES_TO_CHECK))
+    if (png_sig_cmp(buf, 0, PNG_BYTES_TO_CHECK))
     {
         DDebug() << k_funcinfo << "Not a PNG image file." << endl;
         fclose(f);
@@ -252,7 +252,8 @@
 #ifdef ENABLE_DEBUG_MESSAGES
                 DDebug() << "PNG in PNG_COLOR_TYPE_GRAY" << endl;
 #endif
-                png_set_gray_1_2_4_to_8(png_ptr);
+                png_set_expand_gray_1_2_4_to_8(png_ptr);
+
                 png_set_gray_to_rgb(png_ptr);
 
                 if (QImage::systemByteOrder() == QImage::LittleEndian)       // Intel
@@ -659,7 +660,7 @@
                 png_text comment;
                 comment.key         = "Comment";
                 comment.text        = ba.data();
-                comment.itxt_lenght = ba.size();
+                comment.itxt_length = ba.size();
                 comment.compression = PNG_ITXT_COMPRESSION_zTXt;
                 png_set_text(png_ptr, info_ptr, &(comment), 1);
             
--- digikam-0.9.4/digikam/kioslave/digikamthumbnail.cpp~	2008-07-16 22:05:55.000000000 +0300
+++ digikam-0.9.4/digikam/kioslave/digikamthumbnail.cpp	2010-03-01 11:44:14.677566100 +0200
@@ -363,7 +363,7 @@
     unsigned char buf[PNG_BYTES_TO_CHECK];
 
     fread(buf, 1, PNG_BYTES_TO_CHECK, f);
-    if (!png_check_sig(buf, PNG_BYTES_TO_CHECK))
+    if (png_sig_cmp(buf, 0, PNG_BYTES_TO_CHECK))
     {
         fclose(f);
         return qimage;
@@ -458,7 +458,7 @@
     {
         png_set_gray_to_rgb(png_ptr);
         if (png_get_bit_depth(png_ptr, info_ptr) < 8)
-            png_set_gray_1_2_4_to_8(png_ptr);
+            png_set_expand_gray_1_2_4_to_8(png_ptr);
     }
 
     int sizeOfUint = sizeof(unsigned int);
