--- themedesigner/CMakeLists.txt.orig	2010-11-22 18:47:35.000000000 +0100
+++ themedesigner/CMakeLists.txt	2010-12-03 22:50:54.000000000 +0100
@@ -45,6 +45,7 @@
         ${STRIGI_LIBRARIES}
         ${LCMS_LIBRARIES}
         ${GPHOTO2_LIBRARIES}
+        khtml
 )
 
 IF(KDEPIMLIBS_FOUND)
--- digikam/CMakeLists.txt~	2010-11-22 18:47:39.000000000 +0100
+++ digikam/CMakeLists.txt	2010-12-05 19:36:00.781721964 +0100
@@ -238,12 +238,12 @@
         ${QT_QT3SUPPORT_LIBRARY}
         ${STRIGI_LIBRARIES}
         ${LCMS_LIBRARIES}
-        ${GPHOTO2_LIBRARIES}
         digikamcore
         digikamdatabase
         advancedrename
         queuemanager
         cameragui
+        ${GPHOTO2_LIBRARIES}
 )
 
 IF(KDEPIMLIBS_FOUND)
