--- digikam-2.7.0/extra/kipi-plugins/vkontakte/vkapi.cpp~	2012-07-08 01:24:25.000000000 +0200
+++ digikam-2.7.0/extra/kipi-plugins/vkontakte/vkapi.cpp	2012-07-29 13:30:18.680442679 +0200
@@ -18,6 +18,8 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#include <stdio.h>
+
 #include "vkapi.moc"
 
 // Qt includes
@@ -60,7 +62,7 @@
     return m_accessToken;
 }
 
-#include <stdio.h>
+//#include <stdio.h>
 
 void VkAPI::startAuthentication(bool forceLogout)
 {
