--- digikam-4.13.0/extra/kipi-plugins/CMakeLists.txt~	2017-03-01 08:30:30.000000000 +0100
+++ digikam-4.13.0/extra/kipi-plugins/CMakeLists.txt	2017-03-01 18:54:46.419083660 +0100
@@ -470,10 +470,6 @@
             add_subdirectory(panorama)
         endif()
 
-        if(ImageMagick_FOUND AND QTGSTREAMER_FOUND)
-            add_subdirectory(videoslideshow)
-        endif()
-
         if(NOT WIN32)
             configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/templates/kipi-plugins.lsm.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/kipi-plugins.lsm)
         endif()
