--- digikam-5.6.0/extra/kipi-plugins/sendimages/settingswidget.cpp~	2017-06-13 09:48:26.000000000 +0200
+++ digikam-5.6.0/extra/kipi-plugins/sendimages/settingswidget.cpp	2017-09-12 23:52:48.161863338 +0200
@@ -101,6 +101,7 @@ SettingsWidget::SettingsWidget(QWidget*
     d->mailAgentName->insertItem(EmailSettings::SYLPHEED,      QLatin1String("Sylpheed"));
     d->mailAgentName->insertItem(EmailSettings::SYLPHEEDCLAWS, QLatin1String("Sylpheed-Claws"));
     d->mailAgentName->insertItem(EmailSettings::THUNDERBIRD,   QLatin1String("Thunderbird"));
+    d->mailAgentName->insertItem(EmailSettings::ICEDOVE,       QLatin1String("Icedove"));
 #endif
     d->mailAgentName->setCurrentIndex(EmailSettings::DEFAULT);
     d->mailAgentName->setWhatsThis(i18n("Select your preferred external email program here. "
--- digikam-5.6.0/extra/kipi-plugins/sendimages/sendimages.cpp~	2017-09-12 23:51:31.000000000 +0200
+++ digikam-5.6.0/extra/kipi-plugins/sendimages/sendimages.cpp	2017-09-12 23:54:12.504522808 +0200
@@ -614,6 +614,7 @@ bool SendImages::invokeMailAgent()
 
                 case EmailSettings::NETSCAPE:
                 case EmailSettings::THUNDERBIRD:
+                case EmailSettings::ICEDOVE
                 {
                     QString prog;
 
@@ -621,6 +622,10 @@ bool SendImages::invokeMailAgent()
                     {
                         prog = QLatin1String("netscape");
                     }
+                    else if (d->settings.emailProgram == EmailSettings::ICEDOVE)
+                    {
+                        prog = QLatin1String("icedove");
+                    }
                     else
                     {
                         prog = QLatin1String("thunderbird");
--- digikam-5.6.0/./extra/kipi-plugins/sendimages/emailsettings.h.orig	2017-06-13 09:48:26.000000000 +0200
+++ digikam-5.6.0/./extra/kipi-plugins/sendimages/emailsettings.h	2017-09-12 23:51:31.722786416 +0200
@@ -67,6 +67,7 @@ public:
         NETSCAPE,
         SYLPHEED,
         SYLPHEEDCLAWS,
+        ICEDOVE,
         THUNDERBIRD
     };
 
